### PR TITLE
Clarify usage of general options to be consistent with

### PR DESCRIPTION
**Pull request by KSchopmeyer**
Clarify usage of general options to be consistent withthe general options design document.

1. Created a table of all the general options and clarified the grouping of the general options
2. Further documented the grouping of general options

Cleaned up the definition of what is a pywbemcli startup script.

### CODE REVIEW

File: docs/pywbemcli/commands.rst [hunk 1]
```diff
@@ -2879,15 +2879,19 @@ implementations of WBEM servers such as the implementation of OpenPegasus.
 
 The ``connection`` command group includes commands that manage named connection
 definitions that are persisted in a :term:`connections file`.
-This allows maintaining multiple connection definitions and then using any
+This allows maintaining multiple connection :term:1definition`s and then using any
```
Comment (andy-maier) [inline docs/pywbemcli/commands.rst:2882]: Typo "1" in ":term:1definition`"

File: docs/pywbemcli/commands.rst [hunk 2]
```diff
@@ -2879,15 +2879,19 @@ implementations of WBEM servers such as the implementation of OpenPegasus.
...
 one via the :ref:`--name general option`. Only a single connection is
 active (selected) at any point in time but the connection connection can
 be selected on the pywbemcli command line (:ref:`--name general option`) or
 changed within an interactive session using the :ref:`Connection select command`
 
 .. index:: pair: connections file; persistent connection attributes
 
-The attributes of each connection definition in the :term:`connections file` are:
+TODO: define persistent connection attributes or should this be connection definition
```
Comment (andy-maier) [inline docs/pywbemcli/commands.rst:2890]: I guess this is the same. Let's use one consistent term, e.g. "connection definition"

File: docs/pywbemcli/mock_support.rst [hunk 3]
```diff
@@ -90,10 +89,10 @@ The following types of files are supported for the ``--mock-server`` option:
       A flag that contains the value of the boolean
       :ref:`--verbose general option` of pywbemcli.
 
-  * Old-style: The mock script does not have a ``setup()`` function. This
-    approach is not recommended, but it is supported on all supported Python
-    versions. Using old-style mock scripts in a connection definition prevents
-    caching of its mock environment.
+  * Old-style(Python < 3.5): The mock script does not have a ``setup()`` function.
```
Comment (andy-maier) [inline docs/pywbemcli/mock_support.rst:92]: The old style is supported on all Python versions. Only the new style requires Python>=3.5.

**Review by andy-maier — CHANGES_REQUESTED** [review]
See comments

File: docs/pywbemcli/commands.rst (hunk 1 above)
Comment (KSchopmeyer) [reply docs/pywbemcli/commands.rst:2882]: Fixed. DONE

File: docs/pywbemcli/commands.rst (hunk 2 above)
Comment (KSchopmeyer) [reply docs/pywbemcli/commands.rst:2890]: Removed the TODO

File: docs/pywbemcli/mock_support.rst (hunk 3 above)
Comment (KSchopmeyer) [reply docs/pywbemcli/mock_support.rst]: whoops, got a bit carried away.  Fixed DONE

**Review by andy-maier — APPROVED** [review]

**Pull request merged**